File: C++/SieveOfEratosthenes.cpp
// An Algorithm to print all prime numbers less than or equal to a given number.
#include<bits/stdc++.h>
#include<iostream>
using namespace std;
void SieveofEratosthenes(int n){
    int a[n-2],p=2; //a is an array of all integers from 2 to n, p is the iterator
    for(int i=0;i<n-2;i++){
      a[i]=i+2;
    }
    bool prime[n-2]; //true if a number is prime
    for(int i=0;i<n-2;i++){
      prime[i]=true;
    }
    for (int p=2; p*p<=n; p++) {//multiples of p
      if (prime[p] == true) {
        for(int j=p*p;j<=n;j+=p){
          prime[j]=false;
        }  
      }
    }
    for (int p=2; p<=n; p++) 
      if (prime[p]) 
        cout << p <<endl; 
}
int main() {
  int n=20;
  SieveofEratosthenes(n);
  return 0;
}

File: Java/sieveOfEratosthenes.java
import java.util.Scanner;

public class sieveOfEratosthenes {

	public static void main(String[] args) {
		
		Scanner sc = new Scanner(System.in);
		int num = sc.nextInt();
		//Creating a boolean array prime[0..size]
		boolean prime[] = new boolean[num+1];
		
		//Implementing sieve
		for(int i=2;i*i<=num;i++) {
			//If value of prime[i] does not changes then its prime
			if(prime[i]) {
				//Multiples of prime[i] will not be prime
				for(int j=i*i;j<=num;j+=i) {
					prime[j] = false;
				}
			}
		}
		
		//Printing all prime numbers till num
		for(int i=2;i<=num;i++) {
			if(prime[i]) {
				System.out.print(i+" ");
			}
		}
	}
}
